### ISSUE TITLE
Watching for new blocks doesn't fire callback

### ISSUE THREAD

**Issue by ahoppen** [MetaMask/metamask-extension#6381]
**Describe the bug**
When watching for new blocks using `web3.eth.filter('latest', callback)`, the callback is never fired.

**To Reproduce**
1. Unlock MetaMask
2. On any webpage open the console and execute
```javascript
web3 = new Web3(ethereum.provider);
web3.eth.filter('latest', (error, result) => {console.error(error); console.log(result);});
```

**Expected behaviour**
As new blocks are mined a log entry should appear in the console.

**Actual behaviour**
No log entries appear.

**Browser details (please complete the following information):**
 - OS: macOS 10.14.3
 - Browser: Chrome 73.0.3683.86, Firefox 66.0.2 (64-Bit)
 - MetaMask Version: 6.3.1

**Comment by kumavis** [comment]
@ahoppen what exact version of web3js are you using?

**Comment by kumavis** [comment]
can confirm metamask is goofin somewhere

**Comment by kumavis** [comment]
@ahoppen correct syntax is 
```js
web3 = new Web3(ethereum);
```
not 
```js
web3 = new Web3(ethereum.provider);
```

**Comment by kumavis** [comment]
made a repo while exploring this https://github.com/kumavis/metamask-bug-6318-repro

@ahoppen please consider donating some time to contribute to metamask to offset the time i spent debugging your issue

**Comment by ahoppen** [comment]
Ah, I've got no idea, how I managed to get from `web3 = new Web3(ethereum);` to `web3 = new Web3(ethereum.provider);` while reducing the issue. That also explains why the issue suddenly started showing up Ropsten while I previously only experienced it on a local chain. I'm super sorry for that.

I have updated and extended the instruction steps and filed them as a new issue: https://github.com/MetaMask/metamask-extension/issues/6423
